torch/_dynamo/variables/functions.py

```diff
@@ -634,10 +634,17 @@ def as_python_constant(self):
         if self.original:
             return self.original
         else:
+
+            def get_val(v):
+                if isinstance(v, variables.UserDefinedObjectVariable):
+                    return v.value
```

Comment:
Why isn't UDOV defined to be `as_python_constant` returning `value`?

Reply:
Unclear, it should be sound in most cases, but my hunch is that we rely on `as_python_constant` for a lot of routing that maybe isnt always safe? I can look into this more, separately. This should be sound here.

Reply:
Based on our recent discussion about eager guards and lazy VT, the reason this is not sound is because when we create a new VT by accessing the contents of a UDOV, we typically need to install more guards. Allowing the python object to be passed directly would bypass this VT creation.

Reply:
This implies that when you pull out value from a  UserDefinedObjectVariable, you need to define an `ID_MATCH`